packages/flame/lib/src/components/position_component.dart

```diff
@@ -285,6 +285,9 @@ class PositionComponent extends Component {
 
   /// Flip the component horizontally around its center line.
   void flipHorizontallyAroundCenter() {
+    if (anchor.x == 0.5) {
+      return flipHorizontally();
+    }
     final delta = (1 - 2 * _anchor.x) * width;
     transform.x += delta * math.cos(transform.angle);
     transform.y += delta * math.sin(transform.angle);
```

Comment:
```suggestion
    if (_anchor.x != 0.5) {
      final delta = (1 - 2 * _anchor.x) * width;
      transform.x += delta * math.cos(transform.angle);
      transform.y += delta * math.sin(transform.angle);
    }
```
Makes it more clear that this is a simple optimization and not a change in functionality.

Reply:
I'm not sure if this is much clearer, but it is one line less so why not 😄